### PR TITLE
Rename functionConfig for set-labels and update usage doc

### PR DESCRIPTION
**Pull request by mengqiy**
The function is still backward compatible. Old functionConfig
is still accepted.

### CODE REVIEW

File: functions/go/set-labels/README.md [hunk 1]
```diff
@@ -4,29 +4,27 @@
 
 <!--mdtogo:Short-->
 
-Add a list of labels to all resources.
+The set-labels function adds a list of labels to all resources. By default, the
```
Comment (frankfarzan) [inline functions/go/set-labels/README.md:7]: Style guide: set-labels -> `set-labels`

File: tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kpt.dev/v1alpha2
+kind: FunctionResultList
+metadata:
+  name: fnresults
+exitCode: 1
+items:
+  - image: gcr.io/kpt-fn/set-labels:unstable
+    stderr: '[error] /// : failed to configure function: fieldSpecs has been deprecated, please rename it to additionalLabelFields'
```
Comment (frankfarzan) [inline tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml:8]: where are we with this `////` issue?

Comment (frankfarzan) [inline tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml:8]: K6: https://github.com/GoogleContainerTools/kpt/blob/main/docs/style-guides/errors.md

Honestly at this point, the expectation is the style guides are followed without being asked every time. There are a handful of rules.

**Review by frankfarzan — CHANGES_REQUESTED** [review]

File: tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml (hunk 2 above)
Comment (mengqiy) [reply tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml:8]: It has been fixed in https://github.com/kubernetes-sigs/kustomize/pull/3955.
We need to bump kyaml version in functions.

Comment (mengqiy) [reply tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml:8]: https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/382

Comment (mengqiy) [reply tests/set-labels/both-additionalLabelFields-fieldSpecs/.expected/results.yaml]: Thanks for pointing this out.
I usually pay more attention to markdown file and sometimes forget to check the styling in the code (error message). I will start to pay more attention to it too.

**Comment by mengqiy** [comment]
Rebased on top of #382.
Last commit contains new changes.

File: functions/go/set-labels/README.md [hunk 3]
```diff
@@ -4,29 +4,27 @@
 
 <!--mdtogo:Short-->
 
-Add a list of labels to all resources.
+The `set-labels` function adds a list of labels to all resources. By default,
+the function will not only set the labels in `metadata.labels` but also a bunch
+of different places where have references to the labels. These settings are
```
Comment (frankfarzan) [inline functions/go/set-labels/README.md:9]: "bunch of different places" Sounds...imprecise and handwavy ;)
Please rephrase.

File: tests/set-labels/empty-list/simple/.expected/results.yaml [hunk 4]
```diff
@@ -5,8 +5,8 @@ metadata:
 exitCode: 1
 items:
   - image: gcr.io/kpt-fn/set-labels:unstable
-    stderr: '[error] : failed to configure function: function config must be a ConfigMap or SetLabelConfig'
+    stderr: '[error] : failed to configure function: function config must be a `ConfigMap` or `SetLabels`'
```
Comment (frankfarzan) [inline tests/set-labels/empty-list/simple/.expected/results.yaml:8]: it's "functionConfig" not "function config". The former is a specific enttity, the latter is two generic words.

**Comment by mengqiy** [comment]
@frankfarzan @droot PTAL

**Review by frankfarzan — APPROVED** [review]

File: functions/go/set-labels/README.md [hunk 5]
```diff
@@ -4,29 +4,45 @@
 
 <!--mdtogo:Short-->
 
-Add a list of labels to all resources.
+The `set-labels` function adds a list of [labels] to all resources. It's a
+common practice to add a set of labels for all the resource in a package.
+Kubernetes has some [recommended labels].
```
Comment (droot) [inline functions/go/set-labels/README.md:9]: s/all the resources

File: functions/go/set-labels/README.md [hunk 6]
```diff
@@ -4,29 +4,45 @@
...
+
+For example, labels can be used in the following scenarios:
+
+- Identify the KRM resources by querying their labels.
+- Set labels for all resources within a package (e.g. environment=staging).
```
Comment (droot) [inline functions/go/set-labels/README.md:14]: We can point to kubernetes docs at the end of the overview.

You can learn more about labels [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/).

File: functions/go/set-labels/README.md [hunk 7]
```diff
@@ -38,47 +54,45 @@ data:
...
 
-You can use key `fieldSpecs` to specify the resource selector you want to use.
-By default, the function will not only add or update the labels in
-`metadata/labels` but also a bunch of different places where have references to
-the labels. These field specs are defined in
-https://github.com/kubernetes-sigs/kustomize/blob/master/api/konfig/builtinpluginconsts/commonlabels.go#L6
-
-You need to use a custom resource to specify additional information.
+To use a `SetLabels` custom resource as the `functionConfig`, the desired labels
+must be specified in the `labels` field. You can optionally
```
Comment (droot) [inline functions/go/set-labels/README.md:58]: For introducing `additionLabelFields`, we can say something along the lines of:

Sometimes you have resources (especially custom resources) that have labels fields in fields other than the defaults (link-to-common-labels), you can specify such label fields using `additionalLabelFields`.

**Review by droot — CHANGES_REQUESTED** [review]
Looks good, I have some minor suggestions.

**Review by droot — APPROVED** [review]

**Pull request merged**